File: Services/TestLab/pkg/models/testcase.go
package models

type TestCase struct {
	TestCaseID     string     `json:"testCaseId"`
	TestCaseName   string     `json:"testCaseName"`
	TestCaseStatus string     `json:"testCaseStatus"`
	Description    string     `json:"description"`
	Precondition   string     `json:"precondition"`
	ExpectedResult string     `json:"expectedResult"`
	Priority       string     `json:"priority"`
	CreatedAt      string     `json:"createdAt"`
	UpdatedAt      string     `json:"updatedAt"`
	NoOfTestSteps  string     `json:"noOfTestSteps"`
	TestSteps      []TestStep `json:"testSteps"`
}

File: Agent/browser_automation/models/test_case.py
"""
Pydantic models for test case data received from NATS consumer.
"""
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel, Field


class TestStep(BaseModel):
    """Model representing a test step within a test case."""
    stepId: str
    order: str
    description: str
    stepData: str
    stepStatus: str
    createdAt: str
    updatedAt: str


class TestCase(BaseModel):
    """Model representing a test case with its associated metadata and steps."""
    projectId: str
    projectName: str
    projectStatus: str
    moduleId: str
    moduleName: str
    moduleStatus: str
    testCaseId: str
    testCaseName: str
    testCaseStatus: str
    testCasePriority: str
    precondition: Optional[str] = None
    expectedResult: Optional[str] = None
    testSteps: List[TestStep]
    executionDateTime: str
    
    def get_prompts(self) -> List[str]:
        """
        Extract prompts from test steps.
        
        Returns:
            List[str]: List of prompts derived from test step descriptions
        """
        return [step.description for step in self.testSteps if step.description]